### PR TITLE
avoiding NullPointerException agaist ConcurrentHashMap.get(key)

### PR DESCRIPTION
**Pull request by chrisptang**
由于UserController.tokenMap 实例是ConcurrentHashMap，当key为Null时会导致NullPointerException，故而先一步检验authorization不为空

### CODE REVIEW

File: dubbo-admin-server/src/main/java/org/apache/dubbo/admin/controller/MetricsCollectController.java [hunk 1]
```diff
@@ -34,16 +33,13 @@
 import org.apache.dubbo.metadata.definition.model.FullServiceDefinition;
 import org.apache.dubbo.metadata.report.identifier.MetadataIdentifier;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (AlbumenJ) [inline dubbo-admin-server/src/main/java/org/apache/dubbo/admin/controller/MetricsCollectController.java:36]: please install codestyle plugin in your ide
class import with wildcard is not allowed in Dubbo

you can check this in https://github.com/apache/dubbo/blob/master/CONTRIBUTING.md

**Comment by AlbumenJ** [comment]
pls do not change `package.lock` and `pom.xml` with codestyle if you do not change any properties

File: dubbo-admin-server/src/main/java/org/apache/dubbo/admin/controller/MetricsCollectController.java (hunk 1 above)
Comment (chrisptang) [reply dubbo-admin-server/src/main/java/org/apache/dubbo/admin/controller/MetricsCollectController.java]: Ops.. thanks for your comment. changes have been revoked.

**Comment by chrisptang** [comment]
> pls do not change `package.lock` and `pom.xml` with codestyle if you do not change any properties

changes have been revoked

File: dubbo-admin-server/src/main/java/org/apache/dubbo/admin/controller/MetricsCollectController.java [hunk 2]
```diff
@@ -110,7 +112,12 @@ protected void addMetricsConfigToMap(Map<String, String> configMap, String ip) {
             String metaData = providerService.getProviderMetaData(providerIdentifier);
             FullServiceDefinition providerServiceDefinition = new Gson().fromJson(metaData, FullServiceDefinition.class);
             Map<String, String> parameters = providerServiceDefinition.getParameters();
-            configMap.put(parameters.get(Constants.METRICS_PORT), parameters.get(Constants.METRICS_PROTOCOL));
+            if (parameters.containsKey(Constants.METRICS_PORT)) {
+                configMap.put(parameters.get(Constants.METRICS_PORT), parameters.get(Constants.METRICS_PROTOCOL));
+            } else {
+                //using default protocol and port, which is dubbo://20880
+                configMap.put("20880", "dubbo");
+            }
```
Comment (KeRan213539) [inline dubbo-admin-server/src/main/java/org/apache/dubbo/admin/controller/MetricsCollectController.java:120]: 这里是不是应该对前端提示缺少参数,而不是给默认值?

File: dubbo-admin-server/src/main/resources/application.properties [hunk 3]
```diff
@@ -16,9 +16,9 @@
 #
 
 # centers in dubbo2.7
-admin.registry.address=zookeeper://127.0.0.1:2183
-admin.config-center=zookeeper://127.0.0.1:2183
-admin.metadata-report.address=zookeeper://127.0.0.1:2183
+admin.registry.address=zookeeper://127.0.0.1:2181
+admin.config-center=zookeeper://127.0.0.1:2181
+admin.metadata-report.address=zookeeper://127.0.0.1:2181
```
Comment (KeRan213539) [inline dubbo-admin-server/src/main/resources/application.properties:21]: 请不要把本地的改动的配制放到PR里

File: dubbo-admin-server/src/main/resources/logback-spring.xml [hunk 4]
```diff
@@ -16,30 +16,31 @@
...
 
-    <root level="INFO">
-        <appender-ref ref="CONSOLE-LOG-ASYNC"/>
-        <appender-ref ref="FILE-LOG-ASYNC"/>
+    <logger name="org.example.examplewebapp" level="DEBUG"/>
+
+    <root level="info">
+        <appender-ref ref="STDOUT"/>
+        <appender-ref ref="appender.main"/>
     </root>
```
Comment (KeRan213539) [inline dubbo-admin-server/src/main/resources/logback-spring.xml:44]: 请不要把本地的改动放到PR里

File: dubbo-admin-server/src/test/java/org/apache/dubbo/admin/controller/ManagementControllerTest.java [hunk 5]
```diff
@@ -37,158 +38,159 @@
...
-      ConfigDTO configDTO = new ConfigDTO();
-      configDTO.setKey("key" + i);
-      configDTO.setConfig("key1=val1\nkey2=val2");
-      configDTOs.add(configDTO);
-
-      String path = getPath(configDTO.getKey());
-      if (zkClient.checkExists().forPath(path) == null) {
-        zkClient.create().creatingParentsIfNeeded().forPath(path);
-      }
-      zkClient.setData().forPath(path, configDTO.getConfig().getBytes());
```
Comment (KeRan213539) [inline dubbo-admin-server/src/test/java/org/apache/dubbo/admin/controller/ManagementControllerTest.java:138]: 为什么要删除之前的测试?这个PR的改动需要改变原来的测试逻辑吗?

File: dubbo-admin-server/src/test/java/org/apache/dubbo/admin/controller/ManagementControllerTest.java [hunk 6]
```diff
@@ -37,158 +38,159 @@
...
-      ConfigDTO configDTO = new ConfigDTO();
-      configDTO.setKey("shouldDeleteConfigKey" + i);
-      configDTO.setConfig("key1=val1\nkey2=val2");
-      configDTOs.add(configDTO);
-
-      String path = getPath(configDTO.getKey());
-      if (zkClient.checkExists().forPath(path) == null) {
-        zkClient.create().creatingParentsIfNeeded().forPath(path);
-      }
-      zkClient.setData().forPath(path, configDTO.getConfig().getBytes());
```
Comment (KeRan213539) [inline dubbo-admin-server/src/test/java/org/apache/dubbo/admin/controller/ManagementControllerTest.java:166]: 同上

File: dubbo-admin-server/src/test/java/org/apache/dubbo/admin/controller/ManagementControllerTest.java [hunk 7]
```diff
@@ -37,158 +38,159 @@
...
-        }, env, "*"
-    );
-    assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
-    assertThat(responseEntity.getBody(), hasSize(num - 2));
-  }
-
-  private String getPath(String key) {
-    key =  StringUtils.isNotEmpty(key) ? key + Constants.PATH_SEPARATOR  + Constants.DUBBO_PROPERTY : Constants.DUBBO_PROPERTY;
-    return "/dubbo/" + Constants.CONFIG_KEY + Constants.PATH_SEPARATOR + key;
-  }
```
Comment (KeRan213539) [inline dubbo-admin-server/src/test/java/org/apache/dubbo/admin/controller/ManagementControllerTest.java:193]: 同上

File: pom.xml [hunk 8]
```diff
@@ -16,346 +16,346 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.21.0</version>
+            </plugin>
+        </plugins>
+    </build>
 
 
-</project>
+</project>
```
Comment (KeRan213539) [inline pom.xml:361]: pom为什么改动那么多? 请不要随意格式化整个文件,仅对你的改动做格式化就行

**Review by KeRan213539 — CHANGES_REQUESTED** [review]

**Comment by chrisptang** [comment]
@KeRan213539 @AlbumenJ 
thanks for the kind code review.

this pull request contains too many changes that make sense only to my project locally, so i will close this one and open another cleaner request for the same issue 'avoiding NullPointerException against ConcurrentHashMap.get(key)'

**Comment by KeRan213539** [comment]
> @KeRan213539 @AlbumenJ
> thanks for the kind code review.
> 
> this pull request contains too many changes that make sense only to my project locally, so i will close this one and open another cleaner request for the same issue 'avoiding NullPointerException against ConcurrentHashMap.get(key)'

OK, look forward to your contribution~

**Comment by chrisptang** [comment]
> @KeRan213539 @AlbumenJ
> thanks for the kind code review.
> 
> this pull request contains too many changes that make sense only to my project locally, so i will close this one and open another cleaner request for the same issue 'avoiding NullPointerException against ConcurrentHashMap.get(key)'

the new pull request as listed: https://github.com/apache/dubbo-admin/pull/691/commits

**Comment by KeRan213539** [comment]
> > @KeRan213539 @AlbumenJ
> > thanks for the kind code review.
> > this pull request contains too many changes that make sense only to my project locally, so i will close this one and open another cleaner request for the same issue 'avoiding NullPointerException against ConcurrentHashMap.get(key)'
> 
> the new pull request as listed: https://github.com/apache/dubbo-admin/pull/691/commits

OK!

**Pull request closed without merging**